### PR TITLE
PLEASE REVIEW - Dt50 allow unique markdown filenames

### PR DESCRIPTION
**Pull request by coreysals**
Only review the 4 most recent commits:
image

Issue:
https://alliancels.myjetbrains.com/youtrack/issue/DT-50

Instead of the previously mandatory "section.md" name this allows for unique markdown file names for each section in documentation. The unique file name is carried through when converted to html. 

-This makes it easier to search for specific .md files (when uniquely named) and also makes it easier to differentiate multiple section .md files when working/viewing multiple at the same time. 

-"section.md" is still of course an allowed name.

-This does not allow multiple .md's in each section (that could be added later). 

-This does not change the mandatory "layout.yaml" file name in each folder/subfolder.

In order to test this on the documentation-gradle-plugins example project:
-Checkout this branch
-In cmd.exe go to WorkingDirectory/documentation-gradle-plugins/example
-Run "gradlew clean" and/or delete the "build" folder in WorkingDirectory/documentation-gradle-plugins/example
-Run "gradlew --include-build .. aA --rerun-tasks" command
-Run "gradlew --include-build .. combineAll" command
-Check to see that combined documents were generated in the WorkingDirectory/documentation-gradle-plugins/example/documentation/All directory.
-Check to see that generated User Requirements html files are named the same as their corresponding md file names

In order to test this on Midas/Titan documentation:
-Go to gradle.build file for Midas/Titan documentation
-Change the maven-repo path from "release" to "test_DT50_AllowUniqueMarkdownFileNames"
-Rename one/some "section.md" files to "_WhateverYouWant_.md"
-In cmd.exe go to WorkingDirectory/Titan
-Run "gradlew clean" and/or delete the "build" folder in WorkingDirectory/Titan/Doc
-Run "gradlew --refresh-dependencies" command
-Run "gradlew Docs:aA --rerun-tasks" command
-Run "gradlew Docs:combineAll" command
-Check to see that generated html files carry the same name as their corresponding md files (the ones that you changed above)

### CODE REVIEW

File: src/main/groovy/com/alliancels/documentation/FileCopyTask.groovy [hunk 1]
```diff
@@ -33,6 +33,25 @@ class FileCopyTask extends SourceTask {
                 destinationStream << sourceStream
                 sourceStream.close()
                 destinationStream.close()
+                
+                //Create new file path for all images
+                File allImagesPath = new File("${project.buildDir}/documentation/All/Images/")
```
Comment (nicolebartel) [inline src/main/groovy/com/alliancels/documentation/FileCopyTask.groovy:38]: The worry I'd have with this is if two images are named the same. 

We'd maybe want to add to our dev guide an image naming convention to try and prevent that (just an idea).

**Review by jdoutine — APPROVED** [review]
Looks good, works with new user requirements directory I have in place.

**Review by nicolebartel — APPROVED** [review]
I'd say approve as long as we aren't applying this to Midas.  It currently breaks on Midas, but I'm not sure if that's possible unless not linking gradle 0.1.0 prevents that.

File: src/main/groovy/com/alliancels/documentation/FileCopyTask.groovy (hunk 1 above)
Comment (coreysals) [reply src/main/groovy/com/alliancels/documentation/FileCopyTask.groovy:38]: Good catch that's something I didn't think of. We would need unique picture names or else similar named pictures will be overwritten and you'll get picture x where you intended to get picture y.

**Comment by coreysals** [comment]
> I'd say approve as long as we aren't applying this to Midas. It currently breaks on Midas, but I'm not sure if that's possible unless not linking gradle 0.1.0 prevents that.

If we leave Midas on 0.0.14 (or really any version other than the version that would ultimately include these changes when merged/updated) there would be no issue. We can also remove (or fix by giving said file its own appropriate folder so that it is even included in the documentation to begin with) the single file that breaks Midas in the event we do want to use the updated doc tools on Midas in the future.

**Comment by coreysals** [comment]
Thanks for reviewing this has been merged!

**Pull request closed without merging**